src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs

```diff
@@ -409,7 +409,7 @@ public override void VisitIn(QueryExpression fieldName, List<QueryExpression> va
                         {
                             if (item is ValueExpression iv)
                             {
-                                foreach (var id in QueryBuilder.GetValues(_query, _metadata, parameters, iv))
+                                foreach (var id in QueryBuilder.GetValues(_query, _metadata, parameters, iv, distinct:true))
                                 {
                                     AddId(id.Value?.ToString());
```

Comment:
@talweiss1982 @ayende Wouldn't it be better to change the `List<Slice> Ids` to `HashSet<Slice>(SliceComparer.Instance)`?

Reply:
Yes, probably

Reply:
not sure if the logic of the distinct is right for the general case, using an hashset mean we will do distinct on all values coming from different parameters.

Reply:
I think in all cases when we are doing collection query then we want to return distinct results?

Reply:
agreed